### PR TITLE
:pencil: DOP-4098 updates comment

### PR DESCRIPTION
**Pull request by caesarbell**
Part of me searching through all DOP codebases to ensure we are not pulling in `prefix`, `bucket`, and `URL` from repo_branches, I stumbled across an outdated comment, and this PR updates that to reflect that we are now caching from `docsets`

This PR has been updated for updating the Types as well to reflect the changes between`repos_branches` and `docsets`

### CODE REVIEW

**Review by mmeigs — APPROVED** [review]
❤️

**Review by mmeigs — APPROVED** [review]
Seems good to me! Thank you for the clean-up!

File: modules/persistence/src/services/metadata/repos_branches/index.ts [hunk 1]
```diff
@@ -16,17 +16,23 @@ export interface BranchEntry {
...
   project: string;
   branches: BranchEntry[];
-  url: EnvKeyedObject;
-  prefix: EnvKeyedObject;
   internalOnly: boolean;
   [key: string]: any;
 }
 
-const internals: { [key: project]: ReposBranchesDocument } = {};
+export type ReposBranchesDocsetsDocument = ReposBranchesDocument & DocsetsDocument;
```
Comment (seungpark) [inline modules/persistence/src/services/metadata/repos_branches/index.ts:33]: what is this new document type?

IIUC it contains `url, prefix, bucket` but also `repoName, project, ...`

Comment (caesarbell) [reply modules/persistence/src/services/metadata/repos_branches/index.ts:33]: That is correct. The new document type is a combination of `repos_branches` and `docsets`. The following properties come from `docsets` now, which are `prefix`, `bucket` and `URL`.

Comment (seungpark) [reply modules/persistence/src/services/metadata/repos_branches/index.ts:33]: are we constructing a document with both of these properties? seeing that neither `repos_branches` or `docsets` has all of these properties defined. think we should be defining `ReposBranchesDocument` and `DocsetsDocument` separately, not combining them into one. (if connecting the two types, it should be that docsets have a `repos` property that is `ReposBranchesDocument[]` as in the data:

image

image

Comment (mmeigs) [reply modules/persistence/src/services/metadata/repos_branches/index.ts:33]: We're aggregating the two together to have a flat access to all of the properties!

**Pull request merged**